src/Websites/Websites/Utilities/WebsitesClient.cs

```diff
@@ -145,6 +146,7 @@ public void AddCustomHostNames(string resourceGroupName, string location, string
                             {
                                 SiteName = webAppName,
                             });
+                        successCount++;
```

Comment:
I don't think we need to keep track of successcount , the problem here is that the hostname value is an invalid format from the example in the issue, the value is "system.collections.generic.list`1[system.string]" as a result the API fails to add - & throws an exception, as a result when an exception occurs we should stop execution of code or return.

Reply:
I consider the scenario, when they try to add multiple Host Names in a single object and the 1st HN is added successfully where the 2nd HN is failed to add.In this case we should consider either of the following 
1. Remove the successfully added HN and stop the execution. With this the old Host Names will remain as it is.
2. Continue the code execution if at least one custom domain is added and remove the old custom domains.

Here I choose the 2nd option. Please suggest.